cmd/finch/version_test.go

```diff
@@ -76,29 +82,75 @@ func TestPrintVersion(t *testing.T) {
 			ctrl := gomock.NewController(t)
 			logger := mocks.NewLogger(ctrl)
 			lcc := mocks.NewLimaCmdCreator(ctrl)
+			var stdOut bytes.Buffer
+			tc.mockSvc(lcc, logger, ctrl)
 
-			versionActionFunc := func() {
-				require.NoError(t, newVersionAction(lcc, logger).printVersion())
-			}
-		
-			output := captureStdout(t, versionActionFunc)
-			assert.Equal(t, output, version.Version)
-			assert.Equal(t, output, version.GitCommit)
+			assert.NoError(t, newVersionAction(lcc, logger, &stdOut).runAdapter(tc.cmd, tc.args))
 		})
 	}
 }
 
-// func TestVersionAction(t *testing.T) {
-// 	ctrl := gomock.NewController(t)
-// 	logger := mocks.NewLogger(ctrl)
-// 	lcc := mocks.NewLimaCmdCreator(ctrl)
-// 	mockCmd := newVersionCommand(lcc, logger)
+func TestVersionAction_run(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name          string
+		wantErr       error
+		cmd           *cobra.Command
+		mockSvc       func(*mocks.LimaCmdCreator, *mocks.Logger, *gomock.Controller)
+		versionOutput TestFinchVersion
+	}{
+		{
+			name:    "print finch version with Client and Server",
+			wantErr: nil,
+			cmd: &cobra.Command{
+				Use: "version",
+			},
+			mockSvc: func(lcc *mocks.LimaCmdCreator, logger *mocks.Logger, ctrl *gomock.Controller) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Running"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Running")
+
+				command := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("shell", limaInstanceName, "nerdctl", "version", "--format", "json").Return(command)
+				command.EXPECT().Output().Return([]byte(""), nil)
+			},
+			versionOutput: TestFinchVersion{
+				"Client:",
+				"Version:       v1.0.0",
+				"OS/Arch:       linux/amd64",
+				"Git commit:    8ua9sdyas7dg0a78sdbasdasdasd78asudh980b",
```

Comment:
I don't see `tc.versionOutput` being used anywhere currently. Also, not sure if we want to rely on these exact git commit SHAs in our tests. That would mean we have to update these pretty regularly, right? Maybe for these you can just check that they aren't empty unless there's a way to mock them easily?

Reply:
These are randomly generated SHA, not the actual ones. But follows the format.

For the structs, I was thinking doing `assert.Contains(t, tc.versionOutput, stdOut.String()` , but errors out. looking into it